core/src/main/java/org/apache/calcite/rel/metadata/RelMdPredicates.java

```diff
@@ -634,6 +641,126 @@ public RelOptPredicateList getPredicates(RelSubset r,
     return Util.first(list, RelOptPredicateList.EMPTY);
   }
 
+  /**
```

Comment:
is this changed?
github makes it difficult to tell whether the code was just moved.
why move it anyway?

Reply:
I refactored the code to make the changes clearer.